src/toil/jobStores/azureJobStore.py

```diff
@@ -247,12 +246,12 @@ def create(self, jobNode):
         return job
 
     def exists(self, jobStoreID):
-        if self.jobItems.get_entity(row_key=bytes(jobStoreID)) is None:
+        if self.jobItems.get_entity(row_key=str(jobStoreID)) is None:
```

Comment:
I just want to check that my understanding is correct here. jobStoreID is likely a `builtins.str` (i.e. some weird hybrid in py2, and the native `str` in python 2). So, with the removal of the `builtins.str` import from above, we're switching back to the native `str` of each version here and in all similar cases below, right? That is, `row_key` will be a `bytes` (aka Py2 `str`) in Python 2, and a `str` (aka Py2 `unicode`) in Python 3? Is that the intention? (It sounds like a reasonable thing to do if that's what Azure expects, but just want to make sure.)

Reply:
The thing I am trying to deal with is that the Azure API code has type checks in it, so only a 100% genuine `str` will do. No `newstr` or `newbytes` objects, which come from the builtins module in Python 2, will be accepted.

I think that's why we were using `bytes` here before; we were replacing `str` from builtins, but we still had the original `bytes`, which in Python 2 is equal to `str` anyway:
```
>>> type(bytes('abc'))
<type 'str'>
```

We can't do this in Python 3, I don't think, since you can't just go from a string to bytes without encoding, and even if we could I think the Azure API still wants strings.

So I got rid of the str import, and now I am using native str, which converts `newstr` to `str` on Python 2 and does no-op conversions of `str` to `str` on Python 3.

Reply:
I don't think I've actually managed to *run* this test under Python 3; I only have it going on Jenkins under Python 2. I still have to actually set up my local Azure setup.

Reply:
OK, cool. I just wanted to make sure that was intentional, because it seemed a bit weird to me. What a pain that Azure is doing these type checks. (surely they should be checking against `basestring` anyway...)